Modules/posixmodule.c

```diff
@@ -14784,10 +14820,13 @@ os_DirEntry_inode_impl(DirEntry *self)
         wchar_t *path = PyUnicode_AsWideCharString(unicode, NULL);
         Py_DECREF(unicode);
         result = LSTAT(path, &stat);
-        PyMem_Free(path);
 
-        if (result != 0)
-            return path_object_error(self->path);
+        if (result != 0) {
+            path_object_error(self->path);
+            PyMem_Free(path);
+            return NULL;
+        }
+        PyMem_Free(path);
```

Comment:
For me, it's easier to reason about PyMem_Free() when it's closer to where it's used, I would prefer to keep it where it is. In the previous code, for me, the scope of "path" is more obvious to me, thanks to the grouped lines.

Reply:
Done.